### ISSUE TITLE
Wrong Links Value

### ISSUE THREAD

**Issue by dyh135** [pyfa-org/Pyfa#1025]
The EHP after shield harmonizing/Armor Energizing burst are not correct with ingame value, resistance are much higher than ingame value

**Comment by Ebag333** [comment]
do you have numbers from in game, and the eft fit?

**Comment by blitzmann** [comment]
Yeah, gonna need more info than that, ie: the information that it requested when you open a new issue 📦 

EFT fit of the booster and booster, the values that you get vs the values that are expected, etc.

**Comment by dyh135** [comment]
![image](https://cloud.githubusercontent.com/assets/24664128/23534374/7b147aec-ff85-11e6-9197-25f9346a19ab.png)

with only Armor Energizing on, EHP in pyfa 358665

in game value 
![image](https://cloud.githubusercontent.com/assets/24664128/23534401/b6ead39a-ff85-11e6-9c8c-d51481812ba2.png)

EHP 341419

------------------------------------
[Damnation, Link Damnation]

Damage Control II
Corpus X-Type Armor Thermal Hardener
Corpii A-Type Adaptive Nano Plating
Federation Navy 1600mm Steel Plates
Corpus X-Type Armor EM Hardener
Imperial Navy 1600mm Steel Plates

50MN Quad LiF Restrained Microwarpdrive
Small Capacitor Booster II
Sensor Booster II
Republic Fleet Target Painter

Civilian Gatling Railgun
Armor Command Burst II, Rapid Repair Charge
Sisters Expanded Probe Launcher
Armor Command Burst II, Armor Energizing Charge
Armor Command Burst II, Armor Reinforcement Charge
[Empty High slot]
[Empty High slot]

Medium Trimark Armor Pump II
Medium Command Processor I


Curator II x1
Bouncer II x1
Warrior II x10


Targeting Range Script x1
Scan Resolution Script x1
Sisters Combat Scanner Probe x8
Navy Cap Booster 400 x51
ECCM Script x1
Armor Energizing Charge x300
Armor Reinforcement Charge x300
Evasive Maneuvers Charge x298


-------------------------------------------------------------------------

The same issue go with shield harmonizing.

**Comment by Ebag333** [comment]
So this is just using bursts on itself? You aren't using the projected/command tabs?

**Comment by dyh135** [comment]
The issue can be reproduced when it boosts to other shipz, I just make this as a more simple example.

**Comment by blitzmann** [comment]
Ehp is calculated differently in the eve fitting window than it is in pyfa (and virtually all fitting programs). Can look into right now on Mobile, but if the resists are more or less the same, there is probably no issue.

**Comment by dyh135** [comment]
The EHP in eve fitting window is the same as EFT.  The problem was founded when I want to decide whether to fit a damage control or A-type adative membrane, it shows in pyfa that with bouns A-type is better, but actually in game due to stack penalty with boost, damage control is better.

**Comment by Ebag333** [comment]
Pyfa doesn't calculate EHP the same way that in-game or EFT does.

We've known for a long time that there's some rounding going on in EVE (just look a the resists in game).

The million dollar question is, are those resists actually rounded off when the damage taken is calculated, or not?  In my experience, the answer is no, which means that there's a discrepancy between the fitting window and actual damage taken.

All right, math time.  Have an Onyx that Pyfa says has 89.1% resists, in game fitting window says 90%.

03:08:50	Combat	210 - Scourge Fury Heavy Missile - Hits

Tooltip volley: 1851

So for Pyfa, the calcs would be:
1851*0.109 = 201.759

For the fitting window, the calcs would be:
1851*.1 = 185.1

So...they're actually _both wrong_. ☹️ 

In order to get the right numbers, it'd be:
1851*X = 210
X = 0.113452188006483

So translating that to resists (what you see), it'd be:
88.6547811993517%

So Pyfa is off by .445%.  In game window is off by 1.345%

Considering how complex that math is, it doesn't surprise me that it's off, especially if somewhere CCP throws in a `floor()` or similar.

Considering how complex that math is, I'm mildly surprised that we're 3x more accurate than CCPs fitting window (at least in this particular instance).

If I had to guess, CCP is doing a bunch of rounding in the fitting window so you get a nice clean number (`90%`) rather than messy one (`88.6547811993517%`).  That's just a guess though, my experience tells me that that the fitting window and Pyfa are usually not more than 2-3% apart at the most extreme (usually within 1%).

As far as EHP goes, not only is the rounding of resists going to play a part, but EVE doesn't use the same profile that we do.  It used to be your weakest resist, that seems to have changed to something different when Ghost fitting was released.  It's a much better view now than it used to be, but it's still not the same omni profile that we default to.

I can't find a profile that causes the EHP to match what is says in game, so I don't know what they're doing exactly.  I  can say that our numbers appear to be accurate taking the damage profile into account (using our resist numbers).

So the TL;DR: Pyfa's resist numbers aren't perfect, but they're closer to accurate than the in game fitting window because of various rounding being done.  The EHP number differs because of the rounding with resists _and_ different damage profiles being used between the two.
